pkg/plugin/plugin.go

```diff
@@ -437,9 +437,27 @@ func CmdDel(args *skel.CmdArgs) error {
 	if args.Netns == "" {
 		// The CNI_NETNS parameter may be empty according to version 0.4.0
 		// of the CNI spec (https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md).
-		// In accordance with the spec we clean up as many resources as possible.
-		if err := cleanPorts(ovsDriver); err != nil {
-			return err
+		if netconf.DeviceID != "" {
+			// SR-IOV Case - The sriov device is moved into host network namespace when args.Netns is empty.
+			// This happens container is killed due to an error (example: CrashLoopBackOff, OOMKilled)
+			var rep string
+			if rep, err = sriov.GetNetRepresentor(netconf.DeviceID); err != nil {
+				return err
+			}
+			if err = removeOvsPort(ovsDriver, rep); err != nil {
+				// Don't throw err as delete can be called multiple times because of error in ResetVF and ovs
+				// port is already deleted in a previous invocation. so log it and proceed further.
```

Comment:
`so log it and proceed further.` seems superfluous

Reply:
this was needed so that ResetVF is called at line#452 to rename the smartvf back to original name in case ResetVF thrown ErrLinkNotFound error in previous CmdDel call (noticed this behaviour in my tests).

Reply:
Ah, sorry for not being cear enough. I meant the sentence in the comment: `so log it and proceed further.`, I think it's not necessary as it's obvious from the next two lines.

Reply:
ah! yes. is it ok now ?

Reply:
Thanks! Looks good to me